PMMain/Midas/MidasGUI/SCDDigester.cpp

```diff
@@ -106,10 +108,22 @@ void SCDDigester::digestKeyboardData(commandData nextCommand)
             //threadHandle->UpdateGUIToChangeWheels TODO
             scdHandle->setKybdGuiSel(kybdGUISel);
             break;
+
         case kybdGUICmds::SELECT:
+
+            keySelect = scdHandle->getKeySelectAngle() / (360 / kybrdRingData[kybdGUISel].getRingInVectorHandle()->size());
+
+            if (kybdGUISel % 4 == 0)
+            {
+                kybrdCtrl->setKeyChar(kybrdRingData[kybdGUISel].getRingInVectorHandle()->at(keySelect).main);
+            }
+            else
+            {
+                kybrdCtrl->setKeyChar(kybrdRingData[kybdGUISel].getRingOutVectorHandle()->at(keySelect).hold);
+            }
+            kybrdCtrl->sendData();
+
```

Comment:
incorrect because of the %4 misused and this section "select" should only ever use main, not hold